src/Core/Services/RequestValidator.cs

```diff
@@ -359,9 +369,11 @@ public void ValidateInsertRequestContext(InsertRequestContext insertRequestCtx)
         /// <exception cref="DataApiBuilderException"></exception>
         public void ValidateUpsertRequestContext(UpsertRequestContext upsertRequestCtx)
         {
+            string dataSourceName = _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(upsertRequestCtx.EntityName);
+            ISqlMetadataProvider sqlMetadataProvider = _sqlMetadataProviderFactory.GetMetadataProvider(dataSourceName);
```

Comment:
Looks like the way to retrieve sqlMetadataProvider is pretty much the same in all the Validate* functions. Potential to move the common code to a helper function?

Reply:
fixed.